### DISCUSSION TITLE
Weird socket issues. (Error: Error: This socket has been ended by the other party)

### DISCUSSION THREAD

**Discussion by tascord** [PrismarineJS/mineflayer#1973] [Q&A]
Hey!
I've been building a little helper bot on my smp and it's been working right up until about an hour ago.
All of a sudden I'm just getting the error 'Error: This socket has been ended by the other party' thrown at me by the error event on the bot. I haven't introduced anything weird in the bot and I've restarted the server. Also, it appears sometimes that the code just won't log, won't join, and the terminal window will just lock up.

Code:
```
const Client = createBot({
    host: 'localhost',
    username: username,
    password: password
});

Client.on('error', (error) => console.log(`Error:`, error));
```

Error:
```
Error: Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:456:14)
    at Cipheriv.ondata (_stream_readable.js:719:22)
    at Cipheriv.emit (events.js:315:20)
    at addChunk (_stream_readable.js:309:12)
    at readableAddChunk (_stream_readable.js:284:9)
    at Cipheriv.Readable.push (_stream_readable.js:223:10)
    at Cipheriv.Transform.push (_stream_transform.js:166:32)
    at Cipheriv._transform (internal/crypto/cipher.js:135:8)
    at Cipheriv.Transform._read (_stream_transform.js:205:10)
    at Cipheriv.Transform._write (_stream_transform.js:193:12) {
  code: 'EPIPE'
}

```

Minecraft Logs:
```
com.mojang.authlib.GameProfile@1f4dfaef[id=<null>,name=tascord,properties={},legacy=false] (/0:0:0:0:0:0:0:1:64267) lost connection: Internal Exception: io.netty.handler.codec.DecoderException: java.lang.IndexOutOfBoundsException: readerIndex(1) + length(1) exceeds writerIndex(1): PooledUnsafeDirectByteBuf(ridx: 1, widx: 1, cap: 1)
```

**Comment by u9g** [comment]
can you share the code you are running to get this error, and maybe try to get the specific line(s) that make this happen

**Reply by tascord** [reply]
Updated !

**Reply by u9g** [reply]
try manually specifying the version of the server with adding a version option to createbot

**Reply by tascord** [reply]
Alright, now just getting nothing logged at all lol.

**Reply by u9g** [reply]
so does that mean that the bot connected and everything worked? try adding a chat listener and saying something in the server’s chat

**Reply by tascord** [reply]
Nope! Bot didn't join, just nothings being logged. Not my chat logger, login logger or anything.

**Reply by tascord** [reply]
Alright, looks like sometimes it'll log the error above but sometimes just freeze up

**Reply by KaffinPX** [reply]
Try debugging?

**Reply by tascord** [reply]
> Try debugging?

I am not sure how to in this scenario. Nothing is being logged.
